dev/archetypes/maven-java/src/main/resources/archetype-resources/__service1Name__-impl/src/main/java/__packageInPathFormat__/__service1Name__/impl/__service1ClassName__Aggregate.java

```diff
@@ -0,0 +1,124 @@
+package ${package}.${service1Name}.impl;
+
+import akka.cluster.sharding.typed.javadsl.EntityContext;
+import akka.cluster.sharding.typed.javadsl.EntityTypeKey;
+import akka.persistence.typed.PersistenceId;
+import akka.persistence.typed.javadsl.*;
+import com.lightbend.lagom.javadsl.persistence.AkkaTaggerAdapter;
+import ${package}.${service1Name}.impl.${service1ClassName}Command.Hello;
+import ${package}.${service1Name}.impl.${service1ClassName}Command.UseGreetingMessage;
+import ${package}.${service1Name}.impl.${service1ClassName}Event.GreetingMessageChanged;
+
+import java.util.Set;
+
+/**
+* This is an event sourced aggregate. It has a state, {@link ${service1ClassName}State}, which
+* stores what the greeting should be (eg, "Hello").
+* <p>
+* Event sourced aggregate are interacted with by sending them commands. This
+* aggregate supports two commands, a {@link UseGreetingMessage} command, which is
+* used to change the greeting, and a {@link Hello} command, which is a read
+* only command which returns a greeting to the name specified by the command.
+* <p>
+* Commands may emit events, and it's the events that get persisted.
+* Each event will have an event handler registered for it, and an
+* event handler simply applies an event to the current state. This will be done
+* when the event is first created, and it will also be done when the entity is
+* loaded from the database - each event will be replayed to recreate the state
+* of the aggregate.
+* <p>
+* This aggregate defines one event, the {@link GreetingMessageChanged} event,
+* which is emitted when a {@link UseGreetingMessage} command is received.
+*/
+public class ${service1ClassName}Aggregate extends EventSourcedBehaviorWithEnforcedReplies<${service1ClassName}Command, ${service1ClassName}Event, ${service1ClassName}State> {
+
+  public static EntityTypeKey<${service1ClassName}Command> ENTITY_TYPE_KEY =
+    EntityTypeKey
+      .create(${service1ClassName}Command.class, "${service1ClassName}Aggregate");
+
+
+  final private EntityContext<${service1ClassName}Command> entityContext;
+  final private String entityId;
+
+  ${service1ClassName}Aggregate(EntityContext<${service1ClassName}Command> entityContext) {
+    super(
+      PersistenceId.of(
+          entityContext.getEntityTypeKey().name(),
+          entityContext.getEntityId(),
+          // The separator is an optional field. If you want to produce
+          // events compatible with Lagom read-side processors, then the
+          // separator must be an empty String.
+          ""
```

Comment:
@ignasi35, we may want to remove this as well.

Reply:
Yep, I've raised https://github.com/lagom/lagom/pull/2380 porting changes discussed in https://github.com/lagom/lagom-java.g8/pull/70